src/renderer/stores/sequencer.ts

```diff
@@ -76,23 +141,360 @@ export const useSequencerStore = create<State & Actions>()(
     setElements: (val) =>
       set((state) => {
         state.elements = val;
+        if (state.testSequenceDisplayed !== null) {
+          const idx = state.sequences.findIndex(
+            (seq) => seq.project.name === state.testSequenceDisplayed?.name,
+          );
+          if (idx !== -1) {
+            state.sequences[idx].elements = val;
+            state.testSequenceUnsaved = true;
+          }
+        }
+      }),
+    setSequences: (val) =>
+      set((state) => {
+        state.sequences = val;
+        if (state.sequences.length > 0) {
+          // Problem here
+          state.testSequenceDisplayed = state.sequences[0].project;
+          state.testSequenceStepTree = state.sequences[0].tree;
+          state.elements = state.sequences[0].elements;
+          state.testSequenceUnsaved = state.sequences[0].testSequenceUnsaved;
+        } else {
+          state.testSequenceDisplayed = null;
+          state.testSequenceStepTree = {
+            type: "root",
+            children: [],
+            identifiers: [],
+          };
+          state.elements = [];
+          state.cycleConfig = {
+            infinite: false,
+            cycleCount: 1,
+            ptrCycle: -1,
+          };
+          state.testSequenceUnsaved = false;
+        }
       }),
-
     setBackendState: (val) =>
       set((state) => {
-        state.backendState = val;
+        state.playPauseState = val;
+      }),
+    setBackendGlobalState: (val) =>
+      set((state) => {
+        state.backendGlobalState = val;
       }),
     setTestSequenceUnsaved: (val) =>
       set((state) => {
         state.testSequenceUnsaved = val;
       }),
     setTestSequenceTree: (val) =>
       set((state) => {
-        state.testSequenceTree = val;
+        state.testSequenceStepTree = val;
+      }),
+
+    // Sequences ===========================================================
+    updateSequenceStatus: (val: StatusType) =>
+      set((state) => {
+        if (state.testSequenceDisplayed !== null) {
+          const idx = state.sequences.findIndex(
+            (seq) => seq.project.name === state.testSequenceDisplayed?.name,
+          );
+          if (idx === -1) {
+            return;
+          }
+          state.sequences[idx].status = val;
+          if (val === "aborted") {
+            state.sequences[idx].elements.forEach((el) => {
+              if (
+                el.type === "test" &&
+                (el.status === "running" || el.status === "paused")
+              ) {
+                el.status = "aborted";
+              }
+            });
+          }
+        }
+      }),
```

Comment:
can you avoid putting the entire logic of the function inside of the `set` callback?
similarly to try/catch, only put stuff that actually mutates the state inside of `set` (here you have extra validation logic that doesn't belong in lines 200-205)
you can use `get()` to access the state.
it's better because it clearly marks where mutations are happening so it's easier to read/debug.